### ISSUE TITLE
bucket4j-hazelcast:7.6.0 UAT is getting failed while using mvn test

### ISSUE THREAD

**Issue by shivsrivastava** [bucket4j/bucket4j#463]
Hi Team,

I am making the build with UAT enabled of submodule bucket4j-hazelcast:7.6.0. I am using the below version.

Maven command - mvn test
Maven version I am trying - 3.3.1 and above.
JDK version- 8 , 11 and 17

Let me know which java version , Maven version and test command it will work?
I have also included log. Error we are getting.
[build-bucket4j-hazelcast-7.6.0-2024-02-28-06-09-59-242-6273909608.txt](https://github.com/bucket4j/bucket4j/files/14695320/build-bucket4j-hazelcast-7.6.0-2024-02-28-06-09-59-242-6273909608.txt)

_"[14 15:47:08.368] | [INFO] Running io.github.bucket4j.BlockingStrategyTest
[14 15:47:10.402] | [INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.029 s - in io.github.bucket4j.BlockingStrategyTest
[14 15:47:10.403] | [INFO] Running io.github.bucket4j.util.concurrent.BatchHelperSpecification
[14 15:47:10.630] | [ERROR] Tests run: 8, Failures: 0, Errors: 7, Skipped: 0, Time elapsed: 0.222 s <<< FAILURE! - in io.github.bucket4j.util.concurrent.BatchHelperSpecification
[14 15:47:10.630] | [ERROR] test success async case(io.github.bucket4j.util.concurrent.BatchHelperSpecification)  Time elapsed: 0.004 s  <<< ERROR!
[14 15:47:10.630] | java.lang.reflect.InaccessibleObjectException: Unable to make java.util.concurrent.CompletableFuture(java.lang.Object) accessible: module java.base does not "opens java.util.concurrent" to unnamed module @6f29c911
[14 15:47:10.630] | at io.github.bucket4j.util.concurrent.BatchHelperSpecification.test success async _case(BatchHelperSpecification.groovy:27)"__

Please help.

**Comment by vladimir-bukhtoyarov** [comment]
What is UAT?

**Comment by vladimir-bukhtoyarov** [comment]
Should be moved to discussions.

**Comment by shivsrivastava** [comment]
I tried with JDK 8 , 11 and 17 . Do u want specific version? . 
I tried with mvn test command and it got failed.
I want to run test cases of it.

**Comment by vladimir-bukhtoyarov** [comment]
What is UAT?

**Comment by vladimir-bukhtoyarov** [comment]
> I tried with JDK 8 , 11 and 17 . Do u want specific version?

Sorry, but I can not believe in that. Please double check prior writting.  JPMS was even not exists in Java 8. Tests can not fail on Java8, because JPMS was introduced in Java 9. 

As I can see from attachement, only groovy based tests were failed. And only **bucket4j-core** module was run, another submodules was skipped

![image](https://github.com/bucket4j/bucket4j/assets/6761830/7a9b288f-f4a9-4d63-90f2-fc5460d06e50)

So I totally do not understand why issue is named `bucket4j-hazelcast:7.6.0 UAT is getting failed while using mvn test`, there is no problems with Hazelcast, build even not reached this module.

Anyway, errors in your logs are self-describable, each fail is caused by `java.lang.reflect.InaccessibleObjectException`,    for example
![image](https://github.com/bucket4j/bucket4j/assets/6761830/baa3878c-046f-40ab-a939-54657aa2c0c6)
That means that you use Java 17(or higher) where reflection is prohibited for unnamed modules. So you have to use Java 11 to make tests passed.

---------
If you want to stay at java 17 then you need to modify configuration of maven-surefire-plugin there https://github.com/bucket4j/bucket4j/blob/master/bucket4j-parent/pom.xml#L162 by adding following lines:
```xml
<argLine>--add-opens java.base/java.time=ALL-UNNAMED</argLine>
<argLine>--add-opens java.base/java.math=ALL-UNNAMED</argLine>
<argLine>--add-opens java.base/java.lang=ALL-UNNAMED</argLine>
<argLine>--add-opens java.base/java.lang.reflect=ALL-UNNAMED</argLine>
```
But success is not guaranteed because, build was never tests aginst java17

**Comment by shivsrivastava** [comment]
Rechecked , I am using java 8 and tried maven version apache-maven-3.8.5 but UT is getting failed.

**Comment by vladimir-bukhtoyarov** [comment]
> Rechecked , I am using java 8 and tried maven version apache-maven-3.8.5 but UT is getting failed.

Check it again,  InaccessibleObjectException introduced since `Java 9`, it can not be thrown from `Java 8`.

![image](https://github.com/bucket4j/bucket4j/assets/6761830/6aabb251-eb4a-4fca-a891-005487a88d5d)

If you return yet another time without clear context, I will ban you permamently.

**Comment by shivsrivastava** [comment]
> > Rechecked , I am using java 8 and tried maven version apache-maven-3.8.5 but UT is getting failed.
> 
> Check it again, InaccessibleObjectException introduced since `Java 9`, it can not be thrown from `Java 8`.
> 
> ![image](https://private-user-images.githubusercontent.com/6761830/317649485-6aabb251-eb4a-4fca-a891-005487a88d5d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MDU4MTYsIm5iZiI6MTcxMTcwNTUxNiwicGF0aCI6Ii82NzYxODMwLzMxNzY0OTQ4NS02YWFiYjI1MS1lYjRhLTRmY2EtYTg5MS0wMDU0ODdhODhkNWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjlUMDk0NTE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGMyNWM0YmUyMWQ0Yjg1Njg3OGJjMmU2ZGNjMmJhYmViZDRiYzhiODU2ODVkNzBiOGI5ZmU3ZDIyMWUwNGNkNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.bQxXo3TLNQasf-VBeXwbHwlbFR2Vln19iLzxRyKzFXk)
> 
> If you return yet another time without clear context, I will ban you permamently.

Sure . Will check and confirm.

**Comment by shivsrivastava** [comment]
Tried but still got failing.

**Comment by vladimir-bukhtoyarov** [comment]
> Checked and confirmed. and Tried on jdk 8 but still got failed.

This is the last warning prior permament ban. With wich errors build failed? Where is build log? Where is evidance in build log, that build was executed on Java 8?

**Comment by meetankit24** [comment]
At last i also got UT failure with OpenJDK18 and Maven 3.9.6

**Comment by sourabhmagdum45** [comment]
I also got UT failure with OpenJDK17,8 and Maven 3.9.6

**Comment by vladimir-bukhtoyarov** [comment]
> I also got UT failure with OpenJDK17,8 and Maven 3.9.6

@sourabhmagdum45 please do not give up. Try 19, 20, 21 and do not forget about 9,10,11,12,13,14,15,16,17. 
It will be extremelly productive for spending time to build project [that is addressed for 1.8](https://github.com/bucket4j/bucket4j/blob/7.6/bucket4j-parent/pom.xml#L148) on such versions.  Also I will be glad to know who also employed at your company. Let another colleagues do the same and answer to this thread.

**Comment by vladimir-bukhtoyarov** [comment]
Anyway, I am glad to meat with everybody in this thread, and I want to offer to play game that is very popular in Russia. Game is called `Find black cat inside black room`. The cat is `java.lang.reflect.InaccessibleObjectException`. The room is [Openjdk 8 Repository](https://github.com/openjdk/jdk8u-dev). I have a few T-shirt with Bucket4j logo. Guy who first find the cat will win a prize, I will send it by email. I [had tried to find a cat](https://github.com/search?q=repo%3Aopenjdk%2Fjdk8u-dev%20InaccessibleObjectException&type=code), but I failed.

Guys, t-shirt is awaiting you
![image](https://github.com/bucket4j/bucket4j/assets/6761830/3600d6c4-502f-4cae-995a-02225929f02f)
